### DISCUSSION TITLE
Disable torrent storing/downloading

### DISCUSSION THREAD

**Discussion by Baitinq** [KiraLT/torrent-stream-server#221] [Ideas]
Itd be really cool if you could disable the downloading/storing or torrents, that way this awesome project could work in docker containers with very limited space.

Thank you.

**Comment by KiraLT** [comment]
I don't think is possible to disable storing to the disk. We need to store to disk downloaded torrent so we could stream it to the user. So technically you need to have as much disk space as torrents you are streaming at the moment require space. And after you finish streaming and start streaming the new torrent, we can just delete it from the hard.

I was planning to implement a smarted disk management system, but it still will require at least 10 GB disk space (20GB if multiple users will stream different torrents). What disk usage will be best for your case?

**Comment by rudnypc** [comment]
The smart managment is a good idea.

**Comment by pataquets** [comment]
As an idea, if file storage could be abstracted to allow using different backends (disk, S3, etc) for storing torrent pieces, a memory store (Redis, Memcached, etc) backend could be later implemented.
